Microsoft.Toolkit.Uwp.UI.Animations/Extensions/AnimationExtensions.cs

```diff
@@ -96,49 +97,122 @@ public static EasingFunctionBase GetEasingFunction(EasingType easingType)
         /// </summary>
         /// <param name="easingType">The <see cref="EasingType"/> used to generate the CompositionEasingFunction</param>
         /// <param name="compositor">The <see cref="Compositor"/></param>
+        /// <param name="easingMode">The <see cref="EasingMode"/> used to generate the CompositionEasingFunction. Defaults to <see cref="EasingMode.EaseOut"/></param>
         /// <returns><see cref="CompositionEasingFunction"/></returns>
-        public static CompositionEasingFunction GetCompositionEasingFunction(EasingType easingType, Compositor compositor)
+        public static CompositionEasingFunction GetCompositionEasingFunction(EasingType easingType, Compositor compositor, EasingMode easingMode = EasingMode.EaseOut)
         {
             if (DesignTimeHelpers.IsRunningInLegacyDesignerMode)
             {
                 return compositor.CreateLinearEasingFunction();
             }
 
-            return GenerateCompositionEasingFunctionFromEasingType(easingType, compositor);
+            return GenerateCompositionEasingFunctionFromEasingType(easingType, compositor, easingMode);
         }
 
-        private static CompositionEasingFunction GenerateCompositionEasingFunctionFromEasingType(EasingType easingType, Compositor compositor)
+        private static CompositionEasingFunction GenerateCompositionEasingFunctionFromEasingType(EasingType easingType, Compositor compositor, EasingMode easingMode = EasingMode.EaseOut)
         {
             if (easingType == EasingType.Default)
             {
                 easingType = DefaultEasingType;
             }
 
-            switch (easingType)
+#pragma warning disable SA1515 // Single-line comment must be preceded by blank line - suppressed for brevity.
+
+            int key = ((int)easingType * 10) + (int)easingMode;
```

Comment:
This can be accomplish in a more type safe method by creating a static `Dictionary<Tuple<EasingType, EasingMode>, CompositionEasingFunction>`

Reply:
From what I understand this probably won't work due to the way ValueTuples generate their HashCodes (combining each component's HashCode) and they way Enums generate HashCodes (which is effectively just int casting), meaning there'd be quite a number of key collisions :'(

I could potentially convert the names to string but it seems slightly wasteful